applications/StructuralMechanicsApplication/custom_elements/small_displacement_mixed_volumetric_strain_element.cpp

```diff
@@ -1014,6 +1339,69 @@ double SmallDisplacementMixedVolumetricStrainElement::CalculateShearModulus(cons
 /***********************************************************************************/
 /***********************************************************************************/
 
+double SmallDisplacementMixedVolumetricStrainElement::CalculateTau1(
+    const Vector &rVoigtIdAnysotropyMatrixProd,
+    const KinematicVariables &rThisKinematicVariables,
+    const ConstitutiveVariables &rThisConstitutiveVariables,
+    const ProcessInfo &rCurrentProcessInfo) const
+{
+    const auto& r_prop = GetProperties();
+    const auto& r_geometry = GetGeometry();
+    const SizeType dim = r_geometry.WorkingSpaceDimension();
+    const SizeType n_nodes = r_geometry.PointsNumber();
+    const SizeType strain_size = r_prop.GetValue(CONSTITUTIVE_LAW)->GetStrainSize();
+
+    Matrix B_i(strain_size, dim);
+    Matrix aux = ZeroMatrix(dim,dim);
+    for (IndexType i_node = 0; i_node < n_nodes; ++i_node) {
+        for (IndexType k = 0;  k < strain_size; ++k) {
+            for (IndexType l = 0; l < dim; ++l) {
+                B_i(k,l) = rThisKinematicVariables.B(k, i_node * dim + l);
+            }
+        }
+        noalias(aux) += prod(trans(B_i), Matrix(prod(rThisConstitutiveVariables.D, B_i)));
+    }
+    double det;
+    const double c_1 = 1.0;
+    Matrix tau_1_mat(dim, dim);
+    MathUtils<double>::InvertMatrix(aux, tau_1_mat, det);
+    const Vector tau_1_vect = MathUtils<double>::SymmetricTensorToVector(tau_1_mat);
+    const double m_T_tau_1 = c_1*inner_prod(rVoigtIdAnysotropyMatrixProd, tau_1_vect);
+
+    // Uncomment this to use the stabilisation parameters defined in the paper
```

Comment:
Does not work?

Reply:
The reason it is commented

Reply:
Ah. To make switching from one definition to the other more handy, there is no special reason.